docs/extensions/guides/renderer-extension.md

```diff
@@ -379,117 +379,286 @@ The `uninstall()` method is implemented in the example above by utilizing the [`
 
 ``` typescript
-import React from "react"
-import { LensRendererExtension } from "@k8slens/extensions"
-import { myCustomPreferencesStore } from "./src/my-custom-preferences-store"
-import { MyCustomPreferenceHint, MyCustomPreferenceInput } from "./src/my-custom-preference"
-
+import { LensRendererExtension } from "@k8slens/extensions";
+import { ExamplePreference, ExamplePreferenceHint, ExamplePreferenceInput } from "./src/example-preference";
+import { observable } from "mobx";
+import React from "react";
 
 export default class ExampleRendererExtension extends LensRendererExtension {
+
+  @observable preference: ExamplePreference = { enabled: false };
+
   appPreferences = [
     {
-      title: "My Custom Preference",
+      title: "Example Preferences",
       components: {
-        Hint: () => <MyCustomPreferenceHint/>,
-        Input: () => <MyCustomPreferenceInput store={myCustomPreferencesStore}/>
+        Input: () => <ExamplePreferenceInput preference={this.preference}/>,
+        Hint: () => <ExamplePreferenceHint/>
       }
     }
   ];
 }
 ```
 
+App preferences are objects matching the `AppPreferenceRegistration` interface. The `title` field specifies the text to show as the heading on the Preferences page. The `components` field specifies two `React.Component` objects defining the interface for the preference. `Input` should specify an interactive input element for your preference and `Hint` should provide descriptive information for the preference, which is shown below the `Input` element. `ExamplePreferenceInput` expects its React props set to an `ExamplePreference` instance, which is how `ExampleRendererExtension` handles the state of the preference input. `ExampleRendererExtension` has the field `preference`, which is provided to `ExamplePreferenceInput` when it is created. In this example `ExamplePreferenceInput`, `ExamplePreferenceHint`, and `ExamplePreference` are defined in `./src/example-preference.tsx`:
+
+``` typescript
+import { Component } from "@k8slens/extensions";
+import { observer } from "mobx-react";
+import React from "react";
+
+export type ExamplePreference  = {
+  enabled: boolean;
+}
+
+@observer
+export class ExamplePreferenceInput extends React.Component<{preference: ExamplePreference}, {}> {
+
+  render() {
+    const { preference } = this.props;
+    return (
+      <Component.Checkbox
+        label="I understand appPreferences"
+        value={preference.enabled}
+        onChange={v => { preference.enabled = v; }}
+      />
+    );
+  }
+}
+
+export class ExamplePreferenceHint extends React.Component {
+  render() {
+    return (
+      <span>This is an example of an appPreference for extensions.</span>
+    );
+  }
+}
+```
+
```

Comment:
In general, it is useful (especially for reviewing) if every sentence could be on a different line. Markdown on treats double new lines as a paragraph break so this won't break styling.